include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -415,18 +415,13 @@ class transform_iterator
     typedef decltype(__my_unary_func_(::std::declval<typename ::std::iterator_traits<_Iter>::reference>())) reference;
     typedef typename ::std::remove_reference<reference>::type value_type;
     typedef typename ::std::iterator_traits<_Iter>::pointer pointer;
-    typedef typename ::std::random_access_iterator_tag iterator_category;
+    typedef typename ::std::iterator_traits<_Iter>::iterator_category iterator_category;
```

Comment:
So, it would be great to have a test to see if it really works with non-random access iterators (of course on host)

Reply:
Test is added